imager/tests.py

```diff
@@ -0,0 +1,194 @@
+        self.assertIn('Welcome back {}! Continue being awesome!'.format(self.username),
+                      response.content)
+
+    def test_logged_in_home_links(self):
+        response = self.client.get('/')
+        self.assertIn('<a href="/">', response.content)
+        self.assertIn('<a href="/profile/">', response.content)
+        self.assertIn('<a href="/stream/">', response.content)
+        self.assertIn('<a href="/library/">', response.content)
+        self.assertIn('<a href="/accounts/logout/?next=/">', response.content)
+
+    def test_logged_in_home_no_public_photos(self):
+        ImageFactory()
+        response = self.client.get('/')
+        self.assertTemplateUsed(response, template_name='home.html')
+        self.assertIn('imager_images/Space_Needle002.jpg',
+                      response.content)
+
+    def test_logged_in_home_public_photos(self):
+        ImageFactory(published='pub')
+        response = self.client.get('/')
+        self.assertTemplateUsed(response, template_name='home.html')
+        self.assertIn('imager_images/test',
+                      response.content)
+
+    def test_logged_in_profile(self):
+        response = self.client.get('/profile/')
+        self.assertTemplateUsed(response, template_name='profile.html')
+        self.assertIn("{}'s Profile".format(self.username),
+                      response.content)
+
+    def test_logged_in_stream(self):
+        response = self.client.get('/stream/')
+        self.assertTemplateUsed(response, template_name='stream.html')
+        self.assertIn("{}'s Stream".format(self.username),
+                      response.content)
+
+    def test_logged_in_library(self):
+        response = self.client.get('/library/')
+        self.assertTemplateUsed(response, template_name='library.html')
+        self.assertIn("{}'s Library".format(self.username),
+                      response.content)
+
+
+class RegistrationTest(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.new_user = 'username'
+        self.new_password = 'password'
+        self.new_email = 'user@test.com'
+
+    def registration(self):
+        return self.client.post('/accounts/register/',
+                                {'username': self.new_user,
+                                 'password1': self.new_password,
+                                 'password2': self.new_password,
+                                 'email': self.new_email})
+
+    def login_after_registration(self):
+        return self.client.post('/accounts/login/',
+                                {'username': self.new_user,
+                                 'password': self.new_password})
+
+    def test_registration_success(self):
+        response = self.registration()
+        self.assertRedirects(response, '/accounts/register/complete/')
+
+        response = self.client.get('/accounts/register/complete/')
+        self.assertIn('Registration Complete', response.content)
```

Comment:
You should be able to assert here that a registration email was generated.  See the docs on testing Django's email system.